src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -46,7 +46,12 @@ public void OnSearchTextChanged(object sender, EventArgs e)
             if (SearchTextChanged != null)
                 SearchTextChanged(this, e);
 
-            SearchCommand?.Execute(null);
+            // If this is trigged from NodeAutoCompleteSearchViewModel, we don't want to execute this SearchCommand on all nodes.
+            // The Search will later be triggered on a subset of filtered results.
+            if (this is NodeAutoCompleteSearchViewModel is false)
```

Comment:
@QilongTang This was triggering the search command twice that's why we were seeing the results change during the window loading. This is because NodeAutoCompleteSearchViewModel is leveraging the SearchViewModel along with the search text bar. To avoid this, we will not trigger the search on all the nodes in library from NodeAutoCompleteSearchViewModel. Instead we will later search the filtered items based on ML or object type.